scripts/dev-up.sh

```
@@ -0,0 +1,123 @@
+# Find a free port starting from a base value.
+# Checks both host processes (lsof) and Docker port bindings.
+# ---------------------------------------------------------------------------
+find_free_port() {
+  local port=$1
+  while lsof -i :"$port" >/dev/null 2>&1 || docker ps --format '{{.Ports}}' 2>/dev/null | grep -q "0.0.0.0:${port}->"; do
+    port=$((port + 1))
+  done
+  echo "$port"
+}
+
+export APP_PORT=$(find_free_port 5560)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)